### ISSUE TITLE
Passing non existing environmental variable to executable weird behaviour

### ISSUE THREAD

**Issue by friendnick** [matepek/vscode-catch2-test-adapter#121]
When setting `defaultEnv` property with `os_env` that doesn't exist, the environment variable will still be created, with not resolved string value. 

For example
```
"catch2TestExplorer.defaultEnv": {
	"VAR_DOESNT_EXIST": "${os_env:VAR_DOESNT_EXIST}"
}
```
The literal resulting value passed to the executable would be `VAR_DOESNT_EXIST` == 
 `"${os_env:VAR_DOESNT_EXIST}"`.

**Comment by matepek** [comment]
I guess you would prefer and empty string, right?

**Comment by friendnick** [comment]
I think the better approach would be to not create it at all)

**Comment by matepek** [comment]
Would you tell me more about your actual situation you would use it?

**Comment by friendnick** [comment]
The tests are looking for env variables with paths where the test data is stored, in a fallback fasion.
Having "" as a value is no problem, can be recognized as not defined, but more logical would be to not create a variable if it doesn't exist.

**Comment by matepek** [comment]
I can see it is more logical in your case..
But .. considering
- `"PATH": "${os_env:PATH}:${os_env:EXTEND_PATH_WITH_THIS}"`
- `"EXTENDED_VAR": "${os_env:SIMPLE_VAR} is extended with something"`.

In these cases I cannot unset `PATH`. Just wanted to say it is not trivial..

**Comment by friendnick** [comment]
Then having `""` as a value is acceptable solution.

**Comment by matepek** [comment]
Will be fixed in version 2.7.0

**Comment by matepek** [comment]
Note: new var is added `@{os_env_stric:...}`.
